azure-pipelines.yml

```diff
@@ -440,10 +378,99 @@ stages:
+                function Run-Tests-With-Coverage {
+                  param (
+                    $projectPath
+                  )
+                  dotnet test $projectPath --configuration $(BUILD_CONFIGURATION) --results-directory "$(Build.SourcesDirectory)\TestResults" -l trx --no-build --no-restore
+                  Test-ExitCode "ERROR: Unit tests for '$projectPath' FAILED."
+                }
+                dotnet --info
+                Run-Tests-With-Coverage Tests\SonarScanner.MSBuild.Common.Test\SonarScanner.MSBuild.Common.Test.csproj
+                Run-Tests-With-Coverage Tests\SonarScanner.MSBuild.PostProcessor.Test\SonarScanner.MSBuild.PostProcessor.Test.csproj
+                Run-Tests-With-Coverage Tests\SonarScanner.MSBuild.PreProcessor.Test\SonarScanner.MSBuild.PreProcessor.Test.csproj
+                Run-Tests-With-Coverage Tests\SonarScanner.MSBuild.Shim.Test\SonarScanner.MSBuild.Shim.Test.csproj
+                Run-Tests-With-Coverage Tests\SonarScanner.MSBuild.TFS.Test\SonarScanner.MSBuild.TFS.Test.csproj
+                Run-Tests-With-Coverage Tests\SonarScanner.MSBuild.Tasks.IntegrationTest\SonarScanner.MSBuild.Tasks.IntegrationTest.csproj
+                Run-Tests-With-Coverage Tests\SonarScanner.MSBuild.Test\SonarScanner.MSBuild.Test.csproj
+                Run-Tests-With-Coverage Tests\SonarScanner.MSBuild.Tasks.UnitTest\SonarScanner.MSBuild.Tasks.UnitTest.csproj # This one needs to be last to convert the coverage results, see Tests/Directory.Build.targets
+              displayName: 'Run UTs and compute coverage'
+              env:
+                ARTIFACTORY_USER: $(ARTIFACTORY_PRIVATE_READER_USERNAME)
+                ARTIFACTORY_PASSWORD: $(ARTIFACTORY_PRIVATE_READER_ACCESS_TOKEN)
+
+            - task: PublishBuildArtifacts@1
+              inputs:
+                pathToPublish: $(Build.SourcesDirectory)\coverage\
+                artifactName: CoverageReport
+
+            - task: SonarCloudAnalyze@1
+              displayName: 'Code Analysis - End'
+
+            - task: SonarCloudPublish@1
+              displayName: 'Code Analysis - Publish QG'
+              inputs:
+                pollingTimeoutSec: '300'
+
+            - powershell: .\scripts\Mend\Mend-Scan.ps1
+              displayName: "Mend scan"
+              condition: eq(variables.IS_RELEASE_BRANCH, 'true')
+              env:
+                JAVA_HOME_11_X64: '$(JAVA_HOME_11_X64)'
+                WS_PRODUCTNAME: '$(MEND_PRODUCTNAME)'
+                WS_APIKEY: '$(WHITESOURCE_APIKEY)'
+                ARTIFACTORY_USER: $(ARTIFACTORY_PRIVATE_READER_USERNAME)
+                ARTIFACTORY_PASSWORD: $(ARTIFACTORY_PRIVATE_READER_ACCESS_TOKEN)
+
+  - template: stage-with-burgr-notifications.yml@commonTemplates
+    parameters:
+      burgrName: 'qa_it'
+      burgrType: 'qa_it'
+      stageName: 'qa_it'
+      stageDisplayName: 'QA (ITs):'
       stageDependencies:
         - build
       jobs:
```

Comment:
We don't need a new stage.
We can add a new job within the QA stage to split analysis and ITs.

Jobs can be executed in parallel.

```suggestion
```